clikit/src/main/java/org/adamalang/clikit/model/Common.java

```diff
@@ -0,0 +1,95 @@
+package org.adamalang.clikit.model;
+
+import org.adamalang.clikit.exceptions.XMLFormatException;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Common {
+
+    public static Node getFirstNode(NodeList nodeList) {
+        if (nodeList.getLength() == 0)
+            return null;
+        return nodeList.item(0);
+    }
+
+    public static String capitalize(String name) {
+        if (!name.isEmpty())
+            return name.substring(0, 1).toUpperCase() + name.substring(1);
+        return "";
+
+    }
+
+    public static String camelize(String name) {
+        String returnString = "";
+        String[] splitString = name.split("-");
+        for (String word : splitString) {
+            returnString += capitalize(word);
+        }
+        return returnString;
+    }
+
+    public static String escape(String s){
+        Pattern pattern = Pattern.compile("\\s{2,}");
+
+        String returnString = s.replace("\\", "\\\\")
+                .replace("\"", "\\\"")
+                .replace("\n", " ");
+
+        Matcher matcher = pattern.matcher(returnString);
+        returnString = matcher.replaceAll(" ");
+        return returnString;
+    }
+    public static String camelize(String name, boolean firstLower) {
+        if (firstLower) {
+            String returnString = "";
+            String[] splitString = name.split("-");
+            boolean first = true;
```

Comment:
Nit, an interesting way simplify this is with String.join since you have an array
String[] parts = name.split("-");
for (int k = 1; k < parts.length; k++) {
  parts[k] = capitalize(word);
}
return String.join("", parts);